nucypher/cli/commands/alice.py

```diff
@@ -427,23 +434,62 @@ def grant(general_config,
           expiration,
           m, n,
           character_options,
-          config_file):
+          config_file,
+          force):
     """Create and enact an access policy for some Bob. """
 
+    if bob and any((bob_encrypting_key, bob_verifying_key)):
+        message = '--bob canot be used with --bob-encrypting-key or --bob-veryfying key'
+        raise click.BadOptionUsage(option_name='--bob', message=message)
+
     # Setup
     emitter = setup_emitter(general_config)
     ALICE = character_options.create_character(emitter, config_file, general_config.json_ipc)
 
-    # Input validation
+    # Grantee validation
+    if bob:
+        card = Card.load(identifier=bob)
+        bob_verifying_key = card.verifying_key.hex()
+        bob_encrypting_key = card.encrypting_key.hex()
+        emitter.message(f'Loaded card from storage\n'
+                        f'*{card.nickname or card.id.hex()}*\n'
+                        f'Encrypting Key | {card.encrypting_key.hex()}\n'
+                        f'Verifying Key  | {card.verifying_key.hex()}',
+                        color='green')
+        if not force:
+            click.confirm('Is this the correct grantee (Bob)?', abort=True)
+
+    # Policy validation
     if ALICE.federated_only:
         if any((value, rate)):
-            raise click.BadOptionUsage(option_name="--value, --rate",
-                                       message="Can't use --value or --rate with a federated Alice.")
+            message= "Can't use --value or --rate with a federated Alice."
+            raise click.BadOptionUsage(option_name="--value, --rate", message=message)
     elif bool(value) and bool(rate):
         raise click.BadOptionUsage(option_name="--rate", message="Can't use --value if using --rate")
-    elif not (bool(value) or bool(rate)):
+
+    # Policy Shares
+    if not n:
+        n = ALICE.n
+        if not force and not click.confirm(f'Use default value for N ({n})?'):
+            n = click.prompt('Enter total number of shares (N)', type=click.INT)
+    if not m:
+        m = ALICE.m
+        if not force and not click.confirm(f'Use default value for M ({m})?'):
+            n = click.prompt('Enter threshold (M)', type=click.IntRange(1, n+1))
+
+    # Policy Expiration
+    if not force and not expiration:
```

Comment:
Is `expiration` required? If not should it be? - presumably there is no default...

Reply:
Oh is that what we landed on, or is a default expiration/duration still settable from an Alice config?

Reply:
Right, there is a default setting for expiration in Alice's config (`duration_periods`), but where do we get it to use here? In other cases, I see `ALICE.m` or `Alice.N` or `Alice.default_rate`...perhaps I'm missing something.

Reply:
Should we prompt the user to confirm if the default is fine, and if not, then prompt them for an expiration date?